scripts/tests/test_zkai_native_attention_mlp_rmsnorm_label_sensitivity_gate.py

```
@@ -0,0 +1,229 @@
+                    gate.require_output_path(evidence / "out.json")
+
+    def test_write_text_atomically_uses_exclusive_temp_file(self) -> None:
+        with tempfile.TemporaryDirectory() as tmpdir:
+            root = pathlib.Path(tmpdir)
+            evidence = root / "docs" / "engineering" / "evidence"
+            evidence.mkdir(parents=True)
+            output = evidence / "out.json"
+            real_open = gate.os.open
+            temp_flags = []
+
+            def spy_open(path, flags, *args, **kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `spy_open`

(ANN202)